### DISCUSSION TITLE
SURVEY: Keep maintaining the VS Code Extension?  (PLEASE VOTE)

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#813] [Polls]
### With the new twinBASIC IDE becoming more usable, it's time to consider if we should still maintain the old VS Code extension...

**Comment by FullValueRider** [comment]
Realistically, you have to consider the population of VBA developers who use other languages as well.  This prompts the awful suggestion that in addition to maintaining the VSCode extension, you may also need twinBasic extensions for Visual Studio and probably at least a couple of other IDE's.  

Personally I'd like to see the VScode extension maintained until at least rename refactoring and multiple editing windows are implemented (I appreciate that multiple instances of the IDE can be run but this is dangerous if you just want two or more views of the same file for reference purposes)

**Reply by FullValueRider** [reply]
And I suppose my second thought is that the current IDE is only a temporary facility as it is really a stepping stone to an Addin that provides an alternative to the VBA IDE.

**Reply by machthree** [reply]
I've been lurking, watching the progress on this product with a lot of interest. 

> Realistically, you have to consider the population of VBA developers who use other languages as well. 

This is certainly a key point to consider. One of the major attractions of VSCode is the familiarity of the editor for developing in multiple languages/environments. I believe many developers are working in multiple environments and VSCode provides an ecosystem to do that effectively. 

An additional concern is that by not being hostable in VSCode it may reinforce the (incorrect) view that VBx is a legacy/niche product. 

The VSCode marketplace additionally provides a discovery mechanism for the product and a lower resistance way of introducing the product into organisations that use VSCode extensively. 

I can't tell how much of an overhead maintaining both environments is though and it just may be too expensive to do this.

**Reply by bclothier** [reply]
I also agree that keeping VSCode extension would be a good way to attract new developers who otherwise wouldn't have used twinBASIC or has no prior exposure to VBx. I think that it should be possible to support some level of integration between VSCode and the IDE (e.g. launching open IDE to do form editing from VSCode). 

We also need to consider the scenario where multiple languages is needed. Because twinBASIC is relatively new, I predict that we'll have a number of projects where the implementation is in other languages and bindings is provided for twinBASIC to work with. Having VSCode would make those projects easier to work with.

**Comment by DinyaZ** [comment]
Wayne, there are so much things to do to reach the promised full VB6 compatibility and to have a working real-life tool, so it would be only waste of time to split your resources.

Drop VSC, it was only a step in the development.

**Comment by gregse99** [comment]
I actually think that Wayne is the only person who can answer this.  The real question is this: who is the main target audience?  Is it (1) the non-professional users who have learned VBx and have advanced to a point where they want to be able to use the more modern features of programming languages that Microsoft has failed to add since they decided that VBA was already advanced enough for that audience or (2) the professional VBx developer who will now be able to use those modern features or (3) the professional developer who may not have ever considered using VBx but may now be able to so given its advanced features that are on par with other modern languages.  If the answer is 1 and/or 2 then why bother with more than one IDE?  If the answer is 3 then you HAVE to support as many IDEs as necessary for that professional developer audience.

**Reply by EduardoVB** [reply]
I think he could support VSC again once there is a stable version.
Having to build, bugfix and maintain two separate versions now, is a waste of time IMO.

**Reply by bclothier** [reply]
Wayne will also need to consider the marketing aspect --- even if he did almost nothing for VSC add-in, having it visible in the marketplace and accumulating visibility will probably be more efficient when that time comes as opposed to re-starting a dormant product that was not on anybody's radar. Even if the VSC add-in turned into a IDE launcher, that might be enough for the time being as to focus on VB6 compatibility, especially the GUI side of things.

**Reply by DinyaZ** [reply]
> Wayne will also need to consider the marketing aspect --- even if he did almost nothing for VSC add-in, having it visible in the marketplace and accumulating visibility will probably be more efficient when that time comes as opposed to re-starting a dormant product that was not on anybody's radar.


What if Wayne uploads a changelog modification to the Marketplace and gives a link to this github place? New versions can be seen here, says the text.

**Comment by lunasoft2001** [comment]
I think it would be positive for twinBasic to have its extension in vsCode, but as long as this does not prevent compliance with the Roadmap.
I prefer the IDE to be more stable and progress faster.  Once stability is achieved, the vsCode extension can be resumed.  Currently it is missing that the IDE supports the CustomControls with more options (textBox alignment, Button layout, ... ). 
If it works fine in your IDE there is no problem neglecting the extension in vsCode

**Comment by WaynePhillipsEA** [comment]
So the general feeling here seems to be that in the long-term we should try to keep the VS Code extension, but short-term we should freeze maintenance on it until a more appropriate time.   I am in full agreement with that, and that was the outcome I had hoped for.

So indeed, I will make an update to the VS Code extension just to point them to the new IDE information and to explain what's going on.  Thanks all!

**Reply by mansellan** [reply]
I know you plan to open source the IDE at some point - will the same apply to the VSCode extension? I appreciate it's not as simple as just uploading the code to GH.

**Reply by WaynePhillipsEA** [reply]
Yes, both

**Comment by WaynePhillipsEA** [comment]
It's official.  The twinBASIC VS Code extension has had it's last update for 2022:

![image](https://user-images.githubusercontent.com/25738109/165683858-e93b7e76-6219-41b4-b7b3-6c0eb1e80880.png)

**Reply by mansellan** [reply]
I think you've done exactly the right thing. The VS Code extension is still available, still visible in the Marketplace, and still usable as at the version it paused on. Just with a banner to declare the pause.

One question, will VSC Marketplace users still get access to the latest compiler, or is that frozen for them too?

**Reply by WaynePhillipsEA** [reply]
No, they'll be frozen on 0.14.36

**Comment by graysuit** [comment]
<h1>No, drop it</h1>

Even if you put effort in to VScode, there's still limitations.
Off course that's why you made your own IDE.

Leave other IDE and focus on your own IDE (i.e twinbasic).
Then you will get one but powerful IDE.

Also I suggest to rewrite IDE and compiler in twinbasic itself (when TB becomes mature).
Then you will know bugs and problems more effectively.
Also it will prove its maturity.
